crates/stages/stages/src/stages/s3/downloader/worker.rs

```diff
@@ -0,0 +1,110 @@
+use super::error::DownloaderError;
+use reqwest::{header::RANGE, Client};
+use std::path::{Path, PathBuf};
+use tokio::{
+    fs::OpenOptions,
+    io::{AsyncSeekExt, AsyncWriteExt, BufWriter},
+    sync::mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender},
+};
+use tracing::debug;
+
+/// Responses sent by a worker.
+#[derive(Debug)]
+pub(crate) enum WorkerResponse {
+    /// Worker has been spawned and awaiting work.
+    Ready { worker_id: u64, tx: UnboundedSender<WorkerRequest> },
+    /// Worker has downloaded
+    DownloadedChunk { worker_id: u64, chunk_index: usize, written_bytes: usize },
+    /// Worker has encountered an error.
+    Err { worker_id: u64, error: DownloaderError },
+}
+
+/// Requests sent to a worker.
+#[derive(Debug)]
+pub(crate) enum WorkerRequest {
+    /// Requests a range to be downloaded.
+    Download { chunk_index: usize, start: usize, end: usize },
+    /// Signals a worker exit.
+    Finish,
+}
+
+/// Spawns the requested number of workers and returns a `UnboundedReceiver` that all of them will
+/// respond to.
+pub(crate) fn spawn_workers(
+    url: &str,
+    number: u64,
+    data_file: &Path,
+) -> UnboundedReceiver<WorkerResponse> {
+    // Create channels for communication between workers and orchestrator
+    let (orchestrator_tx, orchestrator_rx) = unbounded_channel();
```

Comment:
this is a bit confusing to me,
why do we need multiple workers
doesn't this download just one file